share/availability/discovery/discovery.go

```diff
@@ -11,42 +11,65 @@ import (
 	"github.com/libp2p/go-libp2p/core/network"
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/libp2p/go-libp2p/p2p/host/eventbus"
+	"golang.org/x/sync/errgroup"
 )
 
 var log = logging.Logger("share/discovery")
 
 const (
-	// peerWeight is a weight of discovered peers.
-	// peerWeight is a number that will be assigned to all discovered full nodes,
-	// so ConnManager will not break a connection with them.
-	peerWeight = 1000
-	topic      = "full"
+	topic = "full"
 
 	// eventbusBufSize is the size of the buffered channel to handle
 	// events in libp2p
 	eventbusBufSize = 32
+
+	findPeersStuckWarnDelay = time.Minute
 )
 
 // waitF calculates time to restart announcing.
 var waitF = func(ttl time.Duration) time.Duration {
 	return 7 * ttl / 8
 }
 
+type Parameters struct {
+	// PeersLimit defines the soft limit of FNs to connect to via discovery.
+	// Set 0 to disable.
+	PeersLimit uint
+	// DiscoveryRetryTimeout is an interval between discovery attempts
+	// when we discovered lower than PeersLimit peers.
+	// Set -1 to disable.
+	DiscoveryRetryTimeout time.Duration
```

Comment:
Nit: This setting looks doesn't seem to be very usefull for node runners and could be private to allow to be set only onside code(for tests).